pkg/sql/lease.go

```diff
@@ -562,6 +565,14 @@ type tableState struct {
 		// If set, leases are released from the store as soon as their
 		// refcount drops to 0, as opposed to waiting until they expire.
 		dropped bool
+		// Timer for refreshing the lease before it expires. Once a lease has been
+		// acquired once, a timer continuously runs to refresh the lease on a
+		// separate routine after LeaseRenewalTimeout time.
+		timer *time.Timer
+		// The error that was thrown on the last acquision.
+		// FIXME(joey): Should the first caller return the last async error or
+		// instead try to synchronously get the lease?
+		lastAcquisitionErr error
```

Comment:
I'm still not too sure what to do with the scenario when we get an error during the async acquire. I think the best action, for now, is to ignore the error. (although how do I get around the linter failing because of an unhandled err? D:)

Alternatively, the async refreshing routine could retry but that does get very complicated quickly (I would say out of scope for this PR).

Also, the other suggestion I have in the comment is to return the error on the next blocking acquisition call, but that doesn't make too much sense as that's a past error and doesn't reflect the present (where we may be able to block and succesfully acquire a lease).

Reply:
I think the right approach here is to catch the error and log it using log.Error() that way someone sees it in the logs

Reply:
👍 great, even better!